### PR TITLE
Complete refactoring of task list service and persistency of Task List

### PR DESCRIPTION
**Pull request by rbngz**
Hello @gianlucafrei 
This PR contains a large refactoring of the Task List Service. I completely deleted the `TaskList` domain object and implemented complete all around persistency of Tasks.

### CODE REVIEW

**Comment by rbngz** [comment]
The second part of this PR implements the persistency of the executor pool. Here we only store the static information of the executors to the MongoDB database (E.g., name, id, type..).
When the service is started, the executors are retrieved from the db and stored within the internal repository. This will be used to track assignment of tasks, current state of the executor, etc.) and will be lost once the service stops. (In-memory)

**Comment by rbngz** [comment]
Sorry, forgot to fix the tests. Will do tomorrow evening. Let me know if you want to fix them instead

File: tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorMapper.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package ch.unisg.tapasexecutorpool.pool.adapter.out.persistence.mongodb;
+
+import ch.unisg.tapasexecutorpool.pool.domain.Executor;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (gianlucafrei) [inline tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorMapper.java:6]: Does this class really need to be a bean? I'll add this functionality to the MongoTaskDocument class.

File: tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorPersistenceAdapter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+public class ExecutorPersistenceAdapter implements
+        AddExecutorPort,
+        LoadExecutorPort ,
+        LoadExecutorListPort {
+
+    @Autowired
```
Comment (gianlucafrei) [inline tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorPersistenceAdapter.java:21]: Autowired should no be mixed with constructor injection. Just use one of the two methods for DI.

File: tapas-tasks/src/main/java/ch/unisg/tapastasks/tasks/application/service/NoOpTaskListLock.java [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
 import ch.unisg.tapastasks.tasks.application.port.out.TaskListLock;
-import ch.unisg.tapastasks.tasks.domain.TaskList;
 import org.springframework.stereotype.Component;
 
 @Component
 public class NoOpTaskListLock implements TaskListLock {
     @Override
-    public void lockTaskList(TaskList.TaskListName taskListName) {
+    public void lockTaskList(String taskListName) {
         //do nothing
```
Comment (gianlucafrei) [inline tapas-tasks/src/main/java/ch/unisg/tapastasks/tasks/application/service/NoOpTaskListLock.java:10]: I don't understand this. Why is there a NoOp Lock here.

**Comment by gianlucafrei** [comment]
The tests in the executor pool seem to require the real mongo db. Those tests should run locally without any dependencies.

**Review by gianlucafrei — CHANGES_REQUESTED** [review]

**Comment by gianlucafrei** [comment]
Overall seems really a good refactoring, please just address those minor changes.

File: tapas-tasks/src/main/java/ch/unisg/tapastasks/tasks/application/service/NoOpTaskListLock.java (hunk 3 above)
Comment (rbngz) [reply tapas-tasks/src/main/java/ch/unisg/tapastasks/tasks/application/service/NoOpTaskListLock.java:10]: I wish I would understand this. This file is taken from the tutors repository (See [here](https://github.com/SCS-ASSE-FS21-Tutors/tapas/blob/main/tapas-tasks/src/main/java/ch/unisg/tapastasks/tasks/application/service/NoOpTaskListLock.java)). I thought I just slightly modify this implementation.

File: tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorMapper.java (hunk 1 above)
Comment (rbngz) [reply tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorMapper.java:6]: I did get my inspiration  from the example implementation of the task list persistency. I think it also makes sense to separate this functionality, given our overall granularity and this illustration from the slide decks.
Screenshot 2021-12-12 at 20 46 26

Comment (gianlucafrei) [reply tapas-executor-pool/src/main/java/ch/unisg/tapasexecutorpool/pool/adapter/out/persistence/mongodb/ExecutorMapper.java:6]: Ok agre on the separation part. But does not need to be a bean. Just instantiate it in the Repository class with the constructor. But not a big concern.

**Pull request merged**